common/utils/firebase.py

```diff
@@ -1048,14 +1048,23 @@ def upsert_news(news):
 
 def upsert_praise(praise):
     db = get_db()  # this connects to our Firestore database
-    logger.info(f"Adding praise {praise}")
-
-    #Check if the vibe field exists.
-    if "praise_gif" not in praise:
-        logger.info("vibe field not found, setting praise['vibe'] to None")
-        praise["praise_gif"] = None
-
-    logger.info(f"Adding refined praise {praise} into the database")
+    logger.info(f"Adding praise {praise} into the collections database")  
 
     db.collection("praises").add(praise)
     logger.info("praise successfully saved")
+
+def get_praises_by_user_id(user_id):
+    # Gets all documents with user_id in its database attribute
+    db = get_db()
+    #docs = db.collection('hackathons').where("end_date", ">=", today_str).order_by("end_date", direction=firestore.Query.DESCENDING).stream() 
+    praises= db.collection("praises").where("praise_receiver","==",user_id).get()
+    #praises = db.collection('praises').where("praise_receiver", "==", user_id).order_by("timestamp", direction=firestore.Query.DESCENDING).stream() 
+    # convert each document to a python dictionary
+    praise_list = []
+    for doc in praises:
+        doc_json = doc.to_dict()
+        doc_json["id"] = doc.id
+        praise_list.append(doc_json)
+
+    # return the praise_list sorted in descending order by timestamp
+    return sorted(praise_list, key=lambda x: x["timestamp"], reverse=True)
```

Comment:
Normally you want to rely on the database to sort since it's usually faster than doing it in code